dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/All.java

```diff
@@ -115,14 +115,16 @@ protected void forkRepositoriesFound(Multimap<String, String> pathToDockerfilesI
                 pathToDockerfilesInParentRepo.put(parentRepoName, c.getPath());
                 imagesFoundInParentRepo.put(parentRepoName, image);
 
-                // fork the parent if not already forked or we couldn't fork
-                if (!parentReposAlreadyChecked.contains(parentRepoName)) {
+                // fork the parent if not already forked
+                if (!parentReposForked.contains(parentRepoName)) {
                     GHRepository fork = dockerfileGitHubUtil.closeOutdatedPullRequestAndFork(parent);
                     if (fork == null) {
                         log.info("Could not fork {}", parentRepoName);
                         pathToDockerfilesInParentRepo.remove(parentRepoName, c.getPath());
+                        imagesFoundInParentRepo.remove(parentRepoName, image);
```

Comment:
This seems to be what really affects the logic. It seems like you reverted the name as well as the comment here https://github.com/salesforce/dockerfile-image-update/pull/52/files#diff-850febb926ac48d58dc0962f42249092R118

Reply:
Are you saying with this fix, the logic is being affected?

Reply:
@VijayKumarMidde I don't think renaming `parentReposAlreadyChecked` to `parentReposForked` or putting `parentReposForked.add(parentRepoName)` in the else has any effect. We also lost part of the comment: https://github.com/salesforce/dockerfile-image-update/pull/52/files#diff-850febb926ac48d58dc0962f42249092L118

Reply:
@justinharringa If you don't mind, can you please elaborate why you think putting `parentReposForked.add(parentRepoName)` in the else doesn't have any effect? Thanks!

Reply:
It's not returned and doesn't modify anything that is passed in. What effect did it have moving it into the `else`?

Reply:
Before this change, we used to add repo name to `parentReposForked ` always (ie., even if the fork was not created). 

Say, our `contentsWithImage` array looks like ['abc/Dockerfile', 'abc/Dockerfile'] (if Github returns duplicates) and also assume that repo `abc` is unforkable. In this case, we first add `('abc', 'Dockerfile')` to `pathToDockerfilesInParentRepo` and we fail to fork `abc` repo. So we remove `('abc', 'Dockerfile')` from `pathToDockerfilesInParentRepo`. But we are still adding `abc` to `parentReposForked` set. In the next iteration of for loop we again add `('abc', 'Dockerfile')` to `pathToDockerfilesInParentRepo` and see that `abc` is present in `parentReposForked`. This time we DONT remove `('abc', 'Dockerfile')` from `pathToDockerfilesInParentRepo`. So it goes into infinite loop again.

To fix the above case, I'm adding repo name to `parentReposForked ` only in else condition (ie., only when a fork is created successfully).

Reply:
I should probably add a comment in the code why I'm doing this. let me know if this is clear. I will add some comments to code and update the PR.

Reply:
Or I could simply add to `pathToDockerfilesInParentRepo` and `imagesFoundInParentRepo` only if `fork` is not `null`.

Reply:
@VijayKumarMidde yeah, I'm in favor of simplifying the logic here and getting rid of the `remove()`s. Does `forkRepositoriesFoundAndGetPathToDockerfiles_unableToforkRepo` cover the case where GitHub returns the same name twice? You might put that in the name.

Comments would be super helpful. 

Thanks and sorry about the delay on this.

Reply:
Since `imagesFoundInParentRepo` is passed into this function, it makes me nervous that we are mutating it. If the problem is that dupes are showing up, then could we change this to another form of `Map` that had the fork path as the key?

Reply:
@afalko `imagesFoundInParentRepo` is passed as an empty `Map` to this function. This function was updating `imagesFoundInParentRepo` before these changes.

I think we can't use another form of Map with fork path as the Key because, in some cases we may need to update multiple Dockerfiles present in a repo.

Reply:
@afalko seems like some of this should be refactored to remove duplication and move more towards immutable calls. I could see some of these things being pulled into first-class objects rather than passing around a bunch of `MultiMap`s but that might be a tall order for this PR? Definitely open to ideas though.

Reply:
(part of what I meant to say was that the mutation of the `MultiMap` is something that has been here for a while and wasn't introduced by Vijay)

Reply:
Yeh, maybe file an issue for that and move forward here

Reply:
I opened #71